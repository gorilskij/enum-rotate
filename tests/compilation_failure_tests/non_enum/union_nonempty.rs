use derive_enum_rotate::EnumRotate;

#[derive(EnumRotate)]
union Union {
    field1: usize,
    field2: u8,
}
