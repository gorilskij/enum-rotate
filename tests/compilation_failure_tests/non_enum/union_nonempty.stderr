error: Union Union is not an enum, EnumRotate can only be derived for enums
 --> tests/compilation_failure_tests/non_enum/union_nonempty.rs:4:1
  |
4 | / union Union {
5 | |     field1: usize,
6 | |     field2: u8,
7 | | }
  | |_^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/compilation_failure_tests/non_enum/union_nonempty.rs:7:2
  |
7 | }
  |  ^ consider adding a `main` function to `$DIR/tests/compilation_failure_tests/non_enum/union_nonempty.rs`
